### PR TITLE
Table directive

### PR DESCRIPTION
**Pull request by Anchalexcellence**

### CODE REVIEW

File: app/common/tables.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<div class="col-sm-8">
+   <center>
+      <h4>View Poll Data</h4>
+      {{OutletCtrl.outletdata}}
```
Comment (arun-kumar-codes) [inline app/common/tables.html]: what is outletdata? can u use a better name

File: app/common/tables.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   <div class="container">
+      <table class="table table-condensed">
+         <thead>
+            <tr>
+               <th ng-repeat="data in tableheading">{{data}}</th>
+            </tr>
+         </thead>
+         <tbody>
+            <tr>
+               <td>John</td>
```
Comment (arun-kumar-codes) [inline app/common/tables.html]: why is this hardcoded? this should all have ng-repeat

Comment (arun-kumar-codes) [reply app/common/tables.html]: all tables have some action button what about that?

File: app/components/outlets/outletController.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+var app = angular.module('invisionApp')
+app.controller('viewOutletController', function () {
+  url = '/outlet/get/-1/1'
+  this.data = {
+    id:'ID',
```
Comment (arun-kumar-codes) [inline app/components/outlets/outletController.js]: what is all this? why all this is hard coded here

File: app/directives/tableDirective.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+var app = angular.module('invisionApp')
```
Comment (arun-kumar-codes) [inline app/directives/tableDirective.js]: what is invisionApp? name of project is BAT

File: app/directives/tableDirective.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+app.directive('mytable', function () {
+  return {
+    restrict: 'EAC',
+    templateUrl: './common/tables.html',
+    scope: {
+      parameter: '@'
+    },
+    controller: function ($scope) {
+      $scope.tableheading = JSON.parse($scope.parameter)
```
Comment (arun-kumar-codes) [inline app/directives/tableDirective.js]: why JSON.parse is required? data should be sent directly as json

File: app/index.html [hunk 6]
```diff
@@ -23,9 +23,8 @@
       <script src="bower_components/angular-ui-router/release/angular-ui-router.js"></script>
       <script src="bower_components/angular-resource/angular-resource.js"></script>
       <script src="app.js"></script>
-      <script type="text/javascript" src="index.config.js"></script>
-      <script type="text/javascript" src="index.constant.js"></script>
-      <script type="text/javascript" src="common/interceptor.js"></script>
+      <script type="text/javascript" src="components/outlets/outletController.js"></script>
+      <script type="text/javascript" src="directives/tableDirective.js"></script>
       <link rel="stylesheet" type="text/css" href="common/styles.css">
```
Comment (arun-kumar-codes) [inline app/index.html]: there is no styling in this pull request? how is that possible. u have not done any styling for table??

File: app/components/outlets/outletController.js (hunk 3 above)
Comment (Anchalexcellence) [reply app/components/outlets/outletController.js]: sir every table have different table heading so i need to pass data from controller to directive

File: app/common/tables.html (hunk 1 above)
Comment (Anchalexcellence) [reply app/common/tables.html]: sir removed outlet data this i have used to checked whether data is coming to directive or not from the controller.

File: app/common/tables.html (hunk 2 above)
Comment (Anchalexcellence) [reply app/common/tables.html]: sir action button implemented as button will come dynamically from the controller.
Also removed harcoded and will use ng-repeat when i fire api to receive data.

File: app/directives/tableDirective.js (hunk 4 above)
Comment (Anchalexcellence) [reply app/directives/tableDirective.js]: change app name from invisionapp to batApp

File: app/directives/tableDirective.js (hunk 5 above)
Comment (Anchalexcellence) [reply app/directives/tableDirective.js]: Removed JSON.parse and sent data directly as json

File: app/index.html (hunk 6 above)
Comment (Anchalexcellence) [reply app/index.html]: sir that time i had make a directive to render the table heading on the page now i have used styles too.

File: app/common/tables.html (hunk 1 above)
Comment (Anchalexcellence) [inline app/common/tables.html]: kjbkjhijhpiuh

File: app/common/styles.css [hunk 7]
```diff
@@ -1,40 +1,31 @@
...
-}
-#btnDiv {
-    float: left;
-}
-#notFound {
-    text-align: center;
+.headingcolor {
+    color: rgb(74, 74, 74);
+    font-size: 16px;
+    font-family: Roboto;
```
Comment (mageexxcel) [inline app/common/styles.css]: if you are using roboto font in entire project. then simply apply Roboto to * so it gets applied to all elements

File: app/directives/tableDirective.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+var app = angular.module('batApp')
+app.directive('mytable', function () {
```
Comment (mageexxcel) [inline app/directives/tableDirective.js]: use a better name than mytable

File: app/components/outlets/outletView.html [hunk 9]
```diff
@@ -0,0 +1 @@
+<mytable parameter="OutletCtrl.data"></mytable>
```
Comment (mageexxcel) [inline app/components/outlets/outletView.html]: use a better name than parameter

File: app/components/outlets/outletController.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+var app = angular.module('batApp')
+app.controller('viewOutletController', function () {
+  url = '/outlet/get/-1/1'
```
Comment (mageexxcel) [inline app/components/outlets/outletController.js]: all api requests should be in a seperate file 'services' not on controllers.

File: app/common/tables.html [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+      <div style="margin-top: 60px;">
+         <div class="color">{{tabledetails.tabletitle}}</div>
+         <table class="table table-condensed">
+            <thead>
+               <tr>
+                  <th class="headingcolor" ng-repeat="data in tabledetails.values">{{data}}</th>
+                  <th><a>{{tabledetails.alink.Text}}
+                     </a>
+                  </th>
+                  <th  ng-if="data.Type=='SUBMIT'" ng-repeat="data in tabledetails.button">
```
Comment (mageexxcel) [inline app/common/tables.html]: where is body of the table?

Comment (Anchalexcellence) [reply app/common/tables.html]: sir when i recieve data from the server then i use ng-repeat inthe body  just i have no data to render on the table body

File: app/common/styles.css (hunk 7 above)
Comment (Anchalexcellence) [reply app/common/styles.css]: uses roboto inside the body.

File: app/directives/tableDirective.js (hunk 8 above)
Comment (Anchalexcellence) [reply app/directives/tableDirective.js]: uses dynamicTable in place of mytable

File: app/components/outlets/outletController.js (hunk 10 above)
Comment (Anchalexcellence) [reply app/components/outlets/outletController.js]: yes sir i will make a seprate file for it

File: app/components/outlets/outletView.html (hunk 9 above)
Comment (Anchalexcellence) [reply app/components/outlets/outletView.html]: uses tabledata in place of parameter

File: app/common/tables.html (hunk 11 above)
Comment (mageexxcel) [reply app/common/tables.html]: yes but as per your directive, use tbody is empty??

if data is sent via directive how will u display it.

File: app/app.js [hunk 12]
```diff
@@ -1 +1,11 @@
-var app = angular.module('invisionApp', ['ui.router', 'ngResource']);
+var app = angular.module('batApp', ['ui.router', 'ngResource'])
+app.config(function ($stateProvider, $urlRouterProvider) {
+  $urlRouterProvider.otherwise('/outletViewTable')
+  $stateProvider
+  .state('outletViewTable', {
+    url: '/outletViewTable',
+    templateUrl: 'components/outlets/outletView.html',
+    controller: 'viewOutletController',
+    controllerAs:'OutletCtrl'
```
Comment (mageexxcel) [inline app/app.js]: use single syntax of controllerAs in full app

File: app/common/ajaxRequest.factory.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+var app = angular.module('batApp')
+app.factory('getDataFactory', function (configuration, $resource) {
+  return {
+    getData: function (url) {
```
Comment (mageexxcel) [inline app/common/ajaxRequest.factory.js]: getData ?? which data?

File: app/common/interceptor.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+var app = angular.module('batApp')
+app.factory('Interceptor', function () {
+  return {
+    request: function (config) {
+      config.url = config.url + '?accessToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbiI6NywiaWF0IjoxNDkyNzU3ODY3LCJleHAiOjE0OTI3NjE0Njd9.0rvDQSolKHmYYhQYj_x_1ggaN-OeDjMUVVyFdAd-xzk'
```
Comment (mageexxcel) [inline app/common/interceptor.js]: why is this hard coded

File: app/common/tables.html [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+                  <th></th>
+                  <th  ng-if="data.Type=='SUBMIT'" ng-repeat="data in tabledetails.button">
+                     <button type="submit" class="btn btn-primary"> {{data.Type}}
+                  </th>
+               </tr>
+            </thead>
+            <tbody>
+
+               <tr ng-if="tabledetails.controller=='outletCtrl'" ng-repeat="viewdata in tablebody">
+                  <td>  {{tablebody[$index].value.id}}</td>
```
Comment (mageexxcel) [inline app/common/tables.html]: this should be a generic directive. right now this directive will only work for outlet.

you need to make one single directive for all the tables in the entire app

File: app/components/outlets/outletController.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
+var app = angular.module('batApp')
+app.controller('viewOutletController', function (getDataFactory) {
+  url = '/outlet/get/-1/1'
```
Comment (mageexxcel) [inline app/components/outlets/outletController.js]: what is this

File: app/common/interceptor.js (hunk 14 above)
Comment (Anchalexcellence) [reply app/common/interceptor.js]: sir after login it will save in local storage

File: app/common/ajaxRequest.factory.js (hunk 13 above)
Comment (Anchalexcellence) [reply app/common/ajaxRequest.factory.js]: changed getData to getOutletViewData

File: app/common/tables.html (hunk 15 above)
Comment (Anchalexcellence) [reply app/common/tables.html]: print the value for table using value which will remain same for all table

File: app/components/outlets/outletController.js (hunk 16 above)
Comment (Anchalexcellence) [reply app/components/outlets/outletController.js]: change  url as var url

File: app/common/Interceptor.js [hunk 17]
```diff
@@ -0,0 +1,6 @@
+angular.module('batApp').factory('Interceptor', () => ({
+  request(config) {
+    config.url += '?accessToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0b2tlbiI6NywiaWF0IjoxNDkzMDM1NTM5LCJleHAiOjE0OTMwMzkxMzl9.P9kHC3GK-06JUXam5aS-CLd5vs8kfXP5eGQjyZVYGs0';
```
Comment (mageexxcel) [inline app/common/Interceptor.js]: ???

File: app/common/getDataFactory.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+angular.module('batApp').factory('getDataFactory', (configuration, $resource) => ({
+  getOutletViewData(url) {
+    return $resource(configuration.apihost + url);
```
Comment (mageexxcel) [inline app/common/getDataFactory.js]: url should not be a parameters. actual url should come in this function

File: app/index.config.js [hunk 19]
```diff
@@ -0,0 +1,3 @@
+angular.module('batApp').config(function ($httpProvider) {
+  $httpProvider.interceptors.push('Interceptor')
```
Comment (mageexxcel) [inline app/index.config.js:2]: what is use of this?

File: app/index.constant.js [hunk 20]
```diff
@@ -0,0 +1,3 @@
+     angular.module('batApp').constant('configuration', {
```
Comment (mageexxcel) [inline app/index.constant.js]: how did this code pass eslint?

File: app/components/outlets/viewOutletController.js [hunk 21]
```diff
@@ -0,0 +1,36 @@
+angular.module('batApp').controller('viewOutletController', function(getDataFactory) {
+    var url = '/outlet/get/-1/1';
```
Comment (mageexxcel) [inline app/components/outlets/viewOutletController.js]: remove url from here and put this in service

File: app/common/Interceptor.js (hunk 17 above)
Comment (Anchalexcellence) [reply app/common/Interceptor.js]: sir after the login i will replace this hard coded access token with a localStorage as i need access token to fetch data so i have hardcoded it.

File: app/common/getDataFactory.js (hunk 18 above)
Comment (Anchalexcellence) [reply app/common/getDataFactory.js]: provided actual url to $resource

File: app/components/outlets/viewOutletController.js (hunk 21 above)
Comment (Anchalexcellence) [reply app/components/outlets/viewOutletController.js]: removed url from controller

File: app/index.config.js (hunk 19 above)
Comment (Anchalexcellence) [reply app/index.config.js:2]: sir from here i  am sending authentication token to fetch the data

File: .eslintrc.js [hunk 22]
```diff
@@ -26,5 +26,4 @@ module.exports = {
         "max-len": [2, 120, 2],
         "object-curly-spacing": [2, "always"]
     },
-    "extends": "plugin:angular/johnpapa"
```
Comment (mageexxcel) [inline .eslintrc.js:29]: why is this removed?

File: app/common/Interceptor.js [hunk 23]
```diff
@@ -0,0 +1,9 @@
+angular.module('batApp')
+ .factory('Interceptor', function ($localStorage) {
+   return {
+     request: function (config) {
+       config.url = config.url + '?accessToken=' + $localStorage.token
```
Comment (mageexxcel) [inline app/common/Interceptor.js]: is $localStorage.token always set. what about when user is not logged in?

File: app/common/getDataFactory.js [hunk 24]
```diff
@@ -1,10 +1,8 @@
...
-            return $resource(configuration.apihost + url)
-        }
-    }
-})
+angular.module('batApp').factory('getDataFactory', (configuration, $resource) => ({
+  getOutletViewData () {
+    return $resource(configuration.apihost + '/outlet/get/-1/1')
+  },
+  login (url) {
+    return $resource(configuration.apihost + url)
```
Comment (mageexxcel) [inline app/common/getDataFactory.js]: again same?? url mistake

File: app/components/login/LoginController.js [hunk 25]
```diff
@@ -1,15 +1,15 @@
...
 angular.module('batApp')
-    .controller('LoginController', function($localStorage, getDataFactory, $state) {
-        var vm = this
-        vm.submit = function(data) {
-            var url = '/admin/login'
-            getDataFactory.login(url).save(data).$promise
+    .controller('LoginController', function ($localStorage, getDataFactory, $state) {
+      var vm = this
+      vm.submit = function (data) {
+        var url = '/admin/login'
```
Comment (mageexxcel) [inline app/components/login/LoginController.js]: wrong

File: .eslintrc.js (hunk 22 above)
Comment (Anchalexcellence) [reply .eslintrc.js:29]: sir added this.

File: app/common/Interceptor.js (hunk 23 above)
Comment (Anchalexcellence) [reply app/common/Interceptor.js]: if $localStorage.token have some value then access token will be added to configurl

File: app/common/getDataFactory.js (hunk 24 above)
Comment (Anchalexcellence) [reply app/common/getDataFactory.js]: send url directly to $resource

File: app/components/login/LoginController.js (hunk 25 above)
Comment (Anchalexcellence) [reply app/components/login/LoginController.js]: Removed url from controller

File: app/common/Interceptor.js [hunk 26]
```diff
@@ -0,0 +1,13 @@
+angular.module('batApp')
+ .factory('Interceptor', function ($localStorage) {
+   return {
+     request: function (config) {
+       if ($localStorage.token) {
+         config.url = config.url + '?accessToken=' + $localStorage.token
+       } else {
+         config.url = config.url
```
Comment (mageexxcel) [inline app/common/Interceptor.js]: there no point of this line. it doesn't do anything

Comment (Anchalexcellence) [reply app/common/Interceptor.js]: sir if $localStorage have some value then only it will add access token to it
